nvme-print-json.c

```diff
@@ -4787,3 +4786,34 @@ struct print_ops *nvme_get_json_print_ops(nvme_print_flags_t flags)
 	json_print_ops.flags = flags;
 	return &json_print_ops;
 }
+
+void obj_add_byte_array(struct json_object *o, const char *k, unsigned char *buf, int len)
+{
+	int i;
+
+	_cleanup_free_ char *value = NULL;
+
+	if (!o || !k || !buf || !len)
+		return;
+
+	value = calloc(1, (len + 1) * 2 + 1);
+
+	if (!value) {
+		obj_add_str(o, k, "Could not allocate string");
+		return;
+	}
+
+	sprintf(value, "0x");
+	for (i = 0; i < len; i++)
+		sprintf(&value[strlen(value)], "%02x", buf[len - i - 1]);
```

Comment:
This is a quadratic algorithm for what should be linear. Can you just keep track of the current length in a local variable?

Reversing the bytes also seems a little weird, but I guess I don't know what the intended use case is.

Reply:
> This is a quadratic algorithm for what should be linear. Can you just keep track of the current length in a local variable?

Just fix this as mentioned.

> Reversing the bytes also seems a little weird, but I guess I don't know what the intended use case is.

Unfortunately I do not any idea to resovle this. This outputs the little endian bytes endian as a big endian string as below for example.
```c
buf[0] = 0xaa;
buf[1] = 0xbb;
buf[2] = 0xcc;
buf[3] = 0xdd;
value = "0xbbccbbaa"
```
Thank you for your reviewing.

Reply:
Not really looked into it, so just a stupid comment from my side on the output order. Shouldn't this be output how the spec defines it? IIRC everything as little endian? (if this statement is untrue, I blame my vacation...)

Reply:
Agree with @igaw. It's hard to say without knowing how this function is going to be used. (I would expect the functions to be added in the same patch set as users of the functions.) But for generic "print bytes" functionality, it seems like it would make sense to print the bytes in the order they appear. If the function is interpreting the bytes as an integer, it might make more sense to have "unit" in the function name.

Reply:
Hi,
For example the existing fw-activate-history command outputs the loge page GUID value as below as a big endian string.
```sh
tokunori@tokunori-desktop:~/nvme-cli$ nvme-build ocp fw-activate-history /dev/nvme0
Firmware History Log:
  log identifier:           194
  valid entries:            0
  entries:
  log page version:         1
  log page guid:            0xd11cf3ac8ab24de2a3f6dab4769a796d
```
Also the OCP specification is also describes same as below as D11CF3AC8AB24DE2A3F6DAB4769A796Dh.
![Screenshot from 2024-09-09 23-13-59](https://github.com/user-attachments/assets/b478c14a-d112-4210-a0a5-448d3f5139a8)
So seems the ocp commands just follow the specification descritions.